### ISSUE TITLE
Incorrect balances rendered for custom networks

### ISSUE THREAD

**Issue by fredlacs** [MetaMask/metamask-extension#11126]
**Describe the bug**
When adding custom networks to Metamask, the eth balance for accounts gets rendered incorrectly.

**Steps to reproduce (REQUIRED)**
Steps to reproduce the behavior, libraries used with version number, and/or any setup information to easily reproduce:

MetaMask Version: 9.5.2
Ganache CLI v6.12.2 (ganache-core: 2.13.2)

1. Start a local instance of ganache: `npx ganache-cli --port 8888 --chainId 1212 --mnemonic "horn hammer original lemon chapter weird gun pond fortune blush cupboard cat"`
2. Add it as a custom network to Metamask
3. Import one of the pre-funded accounts, ie using the private key `0xb2c488b68a775c823263a436bbb8876c4ba64c4b21a0713c5fede5ad369ef89b`
4. Metamask extension shows 350965 ETH

**Expected behavior**
It is expected for the user balance to show 100 eth.
Querying ganache's RPC interface returns the correct value: `curl http://localhost:8888 -X POST -H "Content-Type: application/json" -d '{"jsonrpc":"2.0","method":"eth_getBalance","params": ["0x5fe9dD4c80ab7742B62Fb40CE1fBE37D226645A1", "latest"],"id":1}'`

**Screenshots**
![image](https://user-images.githubusercontent.com/32464905/118648758-a8f03480-b7da-11eb-927b-4fb51370a37a.png)

**Browser details (please complete the following information):**
 - OS: OS X
 - Hardware Wallet:
 - Browser: Chrome Version 90.0.4430.212 (Official Build) (x86_64)
 - MetaMask Version: 9.5.2

**Additional context (Error Messages, etc.)**
I was not able to reproduce the error when using ganache at the 8545 port, so the issue seems to be related to custom networks.
I also attempted to intercept the traffic using `mitmproxy` and the metamask extension seems to be receiving the correct eth balance response, but still rendering incorrectly.

**Comment by fetsorn** [comment]
tl;dr; This bug happens if you don't specify the currency symbol of the custom network.

**Ways to reproduce:**
- Have 0.9936 on Matic
- Add matic rpc `https://rpc-mainnet.matic.network` to Metamask. (Or any other rpc)
- Leave the field "Currency symbol (optional)" empty
- Balance of 2115.7335ETH is shown in Metamask
- Edit custom network, specify a currency symbol, e.g. "ETH" (Or any other symbol)
- Balance of 0.9936ETH is shown in Metamask

**Expected behavior**
- Balance of 0.9936 is shown when the currency symbol is not specified

In the original post, if you use ganache at the 8545 port and edit the network to have empty currency symbol field, it renders the wrong balance as well.

**Comment by danjm** [comment]
Need to confirm whether this was fixed
